api/ratelimit/config/ratelimit/v3/rls_conf.proto

```diff
@@ -88,17 +91,4 @@ enum RateLimitUnit {
 
   // The time unit representing a day.
   DAY = 4;
-}
-
-// [#protodoc-title: Rate Limit Config Discovery Service (RLS Conf DS)]
-
-// Return list of all rate limit configs that rate limit service should be configured with.
-service RateLimitConfigDiscoveryService {
-  rpc StreamRlsConfigs(stream envoy.service.discovery.v3.DiscoveryRequest)
-      returns (stream envoy.service.discovery.v3.DiscoveryResponse) {
-  }
-
-  rpc FetchRlsConfigs(envoy.service.discovery.v3.DiscoveryRequest)
-      returns (envoy.service.discovery.v3.DiscoveryResponse) {
-  }
-}
```

Comment:
We need the service definition here. Here [[1]](https://github.com/envoyproxy/go-control-plane/blob/29a4b731c77541ad7d5b31d3c7387820b2bda5b0/ratelimit/service/ratelimit/v3/rls_conf_ds.pb.go#L172) is the generated proto files and it is used in [[2]](https://github.com/envoyproxy/go-control-plane/blob/db2864461c76a66f05e55c31115dd43f296e9d95/pkg/server/v3/server.go#L56)

[1] https://github.com/envoyproxy/go-control-plane/blob/29a4b731c77541ad7d5b31d3c7387820b2bda5b0/ratelimit/service/ratelimit/v3/rls_conf_ds.pb.go#L172
[2] https://github.com/envoyproxy/go-control-plane/blob/db2864461c76a66f05e55c31115dd43f296e9d95/pkg/server/v3/server.go#L56

Reply:
Thank you for your review and comment @renuka-fernando 
As I mentioned in the point 4 of the [PR description](https://github.com/envoyproxy/ratelimit/pull/465#issue-2052970650), is it okay if I separate it out into another file rls_conf_ds.proto like how it was [here before](https://github.com/renuka-fernando/ratelimit/blob/df1e1b41bd571bb82a2ac794284c783fca9fa8e8/api/ratelimit/service/ratelimit/v3/rls_conf_ds.proto)? 
Also, the source of[ rls_conf_ds.pb.go](https://github.com/envoyproxy/go-control-plane/blob/29a4b731c77541ad7d5b31d3c7387820b2bda5b0/ratelimit/service/ratelimit/v3/rls_conf_ds.pb.go#L172) seems to be ratelimit/service/ratelimit/v3/rls_conf_ds.proto instead of the existing [rls_conf.proto](https://github.com/renuka-fernando/ratelimit/blob/main/api/ratelimit/config/ratelimit/v3/rls_conf.proto). 

Please see the following commits:
1. Both rls_conf.proto and rls_conf_ds.proto was added as part of [this commit ](https://github.com/envoyproxy/ratelimit/pull/373/commits/df1e1b41bd571bb82a2ac794284c783fca9fa8e8)
2. Later, they were removed as part of [this commit.](https://github.com/envoyproxy/ratelimit/pull/373/commits/be6c1076fdce97718ac21c4eb3638f187674947a#diff-04987d62d51285909050075cfd5a0567de1de1b7f7330c0c686347e4c7c477ee)
3. In the [final commit ](https://github.com/envoyproxy/ratelimit/pull/373/commits/4f54625c2aadb3f9162b5dc129c6262931939698)rls_conf.proto was added along with service, but the corresponding [go-control-plane's files](https://github.com/envoyproxy/go-control-plane/tree/main/ratelimit) were not updated.

Would like to hear your thoughts on how to proceed here. Thank you

Reply:
Cool. Having a separate file would be fine. I can't exactly remember the reason for adding it to the same file.
Found this comment https://github.com/envoyproxy/go-control-plane/pull/598#discussion_r1008335478 referred in this commit: https://github.com/envoyproxy/ratelimit/pull/371/commits/0428a748fabb382e998af1e828d04bfb35c601f5

Yeah, separating it into another is fine.